src/neovim/nvimUtil.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+import * as vscode from "vscode";
+import { VimState } from "../mode/modeHandler";
+import { Position } from './../common/motion/position';
+import { TextEditor } from "../textEditor";
+import { Configuration } from '../configuration/configuration';
+
+export class Neovim {
+  // Data flows from VS to Vim
+  static async syncVSToVim(vimState: VimState) {
+    const nvim = vimState.nvim;
+    const buf = await nvim.getCurrentBuf();
+    await buf.setLines(0, -1, true, TextEditor.getText().split('\n'));
+
+    await nvim.callFunction("setpos", [".", [0, vimState.cursorPosition.line + 1, vimState.cursorPosition.character, false]]);
+    await nvim.callFunction("setpos", ["'>", [0, vimState.cursorPosition.line + 1, vimState.cursorPosition.character, false]]);
+    await nvim.callFunction("setpos", ["'<", [0, vimState.cursorStartPosition.line + 1, vimState.cursorStartPosition.character, false]]);
```

Comment:
After setting the `'<` and `'>` marks, try `gv` to restore the visual selection. Then when `:` is sent via nvim_input, it will pre-fill `:'<,'>` in the cmdline.